executor/containerdexecutor/executor.go

```diff
@@ -39,10 +40,13 @@ func (w containerdExecutor) Exec(ctx context.Context, meta executor.Meta, root c
 		return err
 	}
 
-	hostsFile, err := oci.GetHostsFile(ctx, w.root)
+	hostsFile, err := oci.GetHostsFile(ctx, w.root, meta.ExtraHosts)
 	if err != nil {
 		return err
 	}
+	if len(meta.ExtraHosts) > 0 {
+		defer os.RemoveAll(hostsFile)
+	}
```

Comment:
The conditional on this cleanup relies on some subtle implementation detail within `oci.GetHostsFile`. It'd be nicer if either the conditional could be removed (perhaps by making the no-extrahosts case make a redundant copy to be removed) or if `oci.GetHostsFile` returned a `func` to do the necessary cleanup.

Reply:
@ijc updated